docassemble/assemblylinewizard/interview_generator.py

```diff
@@ -483,26 +484,49 @@ def source(self, follow_additional_fields=True, document_type="pdf"):
                 content += ")"
               content += "\n" 
             content += "\n"
+            
+        elif self.type == 'main order':
+          lines = [
+            "mandatory: True",
+            "id: main_order_" + self.interview_label,
+            "code: |",
+            "  " + "# Controls the flow of the basic building blocks of the",
+            "  " + "# interview. To use this interview in another interview",
+            "  " + "# delete the `mandatory: True` specifier or this whole block.",
+            "  " + self.intro + "  # Organization intro screen/splash screen",
+            "  " + "# Introduction to this specific interview",
+            "  " + self.interview_label + "_intro",
+            "  " + "# Trigger the whole interview order block to control question order",
+            "  " + self.interview_label,
+            "  " + "signature_date",
+            # Save a snapshot of interview answers. 
+            # We only want a few anonymous variables
+            "  " + "# Save (anonymized) interview statistics.",
+            "  " + "store_variables_snapshot(data={'zip': users[0].address.zip})",
+            "  " + self.interview_label + "_preview_question  # Pre-canned preview screen",
+            "  " + "basic_questions_signature_flow",
+          ];
+          
+          for signature_field in self.signatures:
+            lines.append( "  " + signature_field )
+          lines.append("  " + self.interview_label + "_download")
+          
+          content += '\n'.join(lines) + '\n'
                                    
         elif self.type == 'interview order':
             # TODO: refactor this. Too much of it is assembly-line specific code
             # move into the interview YAML or a separate module/subclass
             content += "id: interview_order_" + self.interview_label + "\n"
             content += "code: |\n"
-            content += "  # This is a placeholder to control logic flow in this interview" + "\n"
-            signatures = set()
+            content += "  # This is a placeholder to control order of questions in this interview\n"
             added_field_names = set()
             for field in self.logic_list:
-              if field.endswith('.signature'):  # save the signatures for the end
-                signatures.add(field)
-              elif not field in added_field_names:
+              if field == 'signature_date' or field.endswith('.signature'):  # signature stuff goes in main block
+                continue
```

Comment:
Not super happy with this way of doing it. Considered making constants of stuff that should be reserved for the main order block or something, but that seemed excessive for two variables. Considered putting other things in constants too, but wanted to hear opinions first.

Reply:
We do have a variable representing signatures: `signature_fields`. It is a list, and we should consider making use of it in the weaver. This code predates that variable.  See https://github.com/SuffolkLITLab/docassemble-MAVirtualCourt/blob/7124077fa400e67febea83ed524f5c6764d3a7b5/docassemble/MAVirtualCourt/data/questions/basic-questions.yml#L926

Reply:
That change could definitely be its own PR. I'll open an issue: #216. Can leave this alone in this PR.